File: ployglots/mta-ployglot-pkg-vault/packages/ts/src/validators.ts
// @ts-nocheck
export class EnvKeyNotFoundError extends Error {
  readonly key: string;

  constructor(key: string) {
    super(`Environment variable '${key}' not found`);
    this.name = 'EnvKeyNotFoundError';
    this.key = key;
    Object.setPrototypeOf(this, EnvKeyNotFoundError.prototype);
  }
}

File: ployglots/mta-ployglot-pkg-vault/packages/py/src/polyglot_vault_file/validators.py
"""Validators — holds EnvKeyNotFoundError."""
from __future__ import annotations


class EnvKeyNotFoundError(Exception):
    """Raised by EnvStore.get_or_throw when a required key is missing."""

    def __init__(self, key: str) -> None:
        self.key = key
        super().__init__(f"Environment variable '{key}' not found")
